### DISCUSSION TITLE
How to get ambiguous name's value when joining

### DISCUSSION THREAD

**Discussion by mdczaplicki** [sqlalchemy/sqlalchemy#11528] [Usage Questions]
Assuming:
* I have two tables that have e.g., `id` column.
* I join those two tables (somehow)
* I query and call `fetchone`
I get a `row: Row` as a result.
I can potentially call `row.id`, but I never know which `id` it corresponds to. How can I distinguish both `id`s?

```
query = select(a_table, b_table).select_from(a_table.join(b_table, a_table.b_id == b_table.id)
result = connection.execute(query)
row = result.fetchone()
row.id
```

**Comment by zzzeek** [comment]
method one:  use `label()`:

     query = select(a_table.c.id.label("a_id"), b_table.c.id.label("b_id"), a_table, b_table)

method two: use [set_label_style](https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.Select.set_label_style) with [LABEL_STYLE_TABLENAME_PLUS_COL](https://docs.sqlalchemy.org/en/20/core/selectable.html#sqlalchemy.sql.expression.SelectLabelStyle.LABEL_STYLE_TABLENAME_PLUS_COL):

     from sqlalchemy import LABEL_STYLE_TABLENAME_PLUS_COL
     query = select(a_table, b_table).set_label_style(LABEL_STYLE_TABLENAME_PLUS_COL)

the docs for set_label_style() are incorrect when they say TABLENAME_PLUS_COL is the default style, the default style is DISAMBIGUATE_ONLY.  that will show up in the site in an hour or two.
